schemes/scheme-model.js

```diff
@@ -0,0 +1,47 @@
+const db = require('../data/db-config');
+
+module.exports = {
+    find,
+    findById,
+    findSteps,
+    update,
+    remove,
+    add
+};
+
+function find() {
+    return db('steps');
+}
+
+function findById(id) {
+    return db('schemes')
+        .where({ id })
+        .first();
+}
+
+function findSteps(userId) {
+    return db('steps as st')
+        .join('schemes as sc', 'sc.id', 'st.scheme_id')
+        .select('st.id', 'sc.scheme_name', 'st.instructions')
+        .where({ 'sc.id': userId })
+}
+
+function add(scheme) {
+    return db('schemes')
+        .insert(scheme)
+        .then((ids => {
+            return findById(ids[0]);
+        }));
+};
+
+function update(id, changes) {
+    return db('schemes')
+        .where({ id })
+        .update(changes)
+}
+
+function remove(id) {
+    return db('schemes')
+      .where('id', Number(id))
```

Comment:
you do not need to Number check the id. use a middleware instead to trap non integer ids